iib/workers/tasks/build.py

```diff
@@ -1431,3 +1466,43 @@ def _add_ocp_label_to_index(ocp_version, temp_dir, dockerfile_name):
         log.debug(
             'Added the following line to %s: %s', dockerfile_name, label,
         )
+
+
+def _add_distribution_scope_label_to_index(distribution_scope, temp_dir, dockerfile_name):
+    """
+    Add the distribution scope to the provided dockerfile.
+
+    :param str distribution_scope: the distribution_scope that the index is for.
+    :param str temp_dir: the temp directory to look for the dockerfile.
+    :param str dockerfile_name: the dockerfile name.
+    """
+    with open(os.path.join(temp_dir, dockerfile_name), 'a') as dockerfile:
+        label = f'LABEL com.redhat.index.delivery.distribution_scope="{distribution_scope}"'
+        dockerfile.write(f'\n{label}\n')
+        log.debug(
+            'Added the following line to %s: %s', dockerfile_name, label,
+        )
+
+
+def _validate_distribution_scope(resolved_distribution_scope, distribution_scope):
+    """
+    Validate distribution scope is allowed to be updated.
+
+    :param str resolved_distribution_scope: the distribution_scope that the index is for.
+    :param str distribution_scope: the distribution_scope that has been requested the index is for.
+    :return: the valid distribution_scope
+    :rtype: str
+    :raises IIBError: if the resolved_distribution_scope is of lesser scope then distribution_scope
+    """
+    resolved_distribution_scope = resolved_distribution_scope.upper()
+    distribution_scope = (distribution_scope or 'PROD').upper()
+
+    if (resolved_distribution_scope == 'STAGE' and distribution_scope == 'PROD') or (
```

Comment:
If there are plans to add more distribution scopes in the future, this should probably be rewritten.

Reply:
I think I agree, do we think this can be a follow-on?

Reply:
You mean change this separately from this PR? I guess we can leave it to later.